### DISCUSSION TITLE
b3dm Runtime import

### DISCUSSION THREAD

**Discussion by dhulsegge** [atteneder/glTFast#457] [General]
Hi, i was wondering if it is possible to import b3dm or the GLTF inside the b3dm with the usage of GLTFast

**Comment by atteneder** [comment]
Hi @dhulsegge ,

As of now that is not supported.

I'm no expert when it comes to b3dm, but I'm quite certain that one can write a b3dm loader that uses glTFast to load the glTF payload within b3dm (granted the content is valid glTF 2.0).

Note: b3dm is becoming obsolete, afaik.

**Reply by dhulsegge** [reply]
Hi atteneder,

thanks for your quick response. I really love your library and i think its the best out there now. 
i found also this:  https://github.com/NASA-AMMOS/Unity3DTiles but it uses UnityGLTF... 
if i check this answer the payload should be a valid GLTF 
https://gis.stackexchange.com/questions/272484/what-is-difference-between-b3dm-and-gltf#:~:text=b3dm%20(%22Batched%203D%20Model%22,render%20it%20on%20a%20client.

**Reply by atteneder** [reply]
Yeah, so it might be as easy as replacing UnityGLTF with glTFast here:

<https://github.com/NASA-AMMOS/Unity3DTiles/blob/master/Assets/Unity3DTiles/B3DMComponent.cs>

> Note: glTFast does not support colliders nor LoDs. The script above suggests UnityGLTF does.

**Reply by dhulsegge** [reply]
I think that should work thanks! i will give it a try!
